src/quartic/pipeline/validator/utils.py

```diff
@@ -0,0 +1,89 @@
+import json
+from collections import defaultdict
+import pprint
+from networkx.drawing.nx_agraph import write_dot
```

Comment:
Agreed, although I think this is fine for now - it *seems* to be OK at least on import. I assume running will error though and agree it shouldn't be offered if it's not installed.